url708d477f92_zSakare_SENG3011_tgz-pJ8-test_runner_TestRunnerJ8/src/main/implementations/OrderBuilderImpl.java

```diff
@@ -0,0 +1,220 @@
+	private String transactionId;
+	private String bidId;
+	private String askId;
+	private boolean isBid;
+	private Date entryTime;
+	private double oldPrice;
+	private int oldVolume;
+	private String buyerBrokerId;
+	private String sellerBrokerId;
+	
+	public OrderBuilderImpl(String security, Date date, double price, String volume, 
+				String isBid, String brokerId) {
+
+		setSecurity(security);
+		setBid(isBid);
+		if (("B").equals(isBid)) {
+			setBuyerBrokerId(brokerId);
+			setSellerBrokerId(DEFAULT_SELLER_BROKER_ID);
+		} else {
+			setSellerBrokerId(brokerId);
+			setBuyerBrokerId(DEFAULT_BUYER_BROKER_ID);
+		}
+		setDateTime(date);
+		setPrice(price);
+		setVolume(volume);
+		setValue(this.price*this.volume);
+		setRecordType(DEFAULT_RECORD_TYPE);
+		setBidId(DEFAULT_BIDID);
+		setAskId(DEFAULT_ASKID);
+		setEntryTime(DEFAULT_ENTRY_TIME);
+		setOldPrice(DEFAULT_OLD_PRICE);
+		setOldVolume(DEFAULT_OLD_VOLUME);
+		setUndisclosedVolume(DEFAULT_UNDISCLOSED_VOLUME);
+		setQualifiers(DEFAULT_QUALIFIERS);
+		
+	}
+	
+	public OrderBuilderImpl(SircaOrder order) {
+		setSecurity(order.getInstrument());
+		setBid(order.getBidOrAsk());
+		setBuyerBrokerId(order.getBuyerBrokerId());
+		setSellerBrokerId(order.getSellerBrokerId());
+		setDateTime(order.getDate(), order.getTime());
+		if (!order.getPrice().isEmpty()) {
+			setPrice(order.getPrice());
+		} else {
+			setPrice(DEFAULT_PRICE);
+		}
+		if (!order.getVolume().isEmpty()) {
+			setVolume(order.getVolume());
+		} else {
+			setVolume(DEFAULT_VOLUME);
+		}
+		setRecordType(order.getRecordType());
+		setUndisclosedVolume(DEFAULT_UNDISCLOSED_VOLUME);
+		setQualifiers(order.getQualifiers());
+		setTransactionId(order.getTransactionId());
+		setBidId(order.getBidId());
+		setAskId(order.getAskId());
+		setEntryTime(DEFAULT_ENTRY_TIME);
+		setOldPrice(DEFAULT_OLD_PRICE);
+		setOldVolume(DEFAULT_OLD_VOLUME);
+		setValue(this.price*this.volume);
+	}
+	
+	public OrderBuilderImpl (String security, Date date, double price, String volume,
+				String brokerID) {
+		
+		setSecurity(security);
+		setDateTime(date);
+		setPrice(price);
+		setVolume(volume);
+		
+	}
+	
+	public Order build() {
+		return new Order(security, 
+							dateTime, 
+							recordType, 
+							price, 
+							volume, 
+							undisclosedVolume, 
+							value, 
+							qualifiers, 
+							transactionId, 
+							bidId, 
+							askId,
+							isBid,
+							entryTime,
+							oldPrice,
+							oldVolume,
+							buyerBrokerId,
+							sellerBrokerId);
+	}
+
+	public void setSecurity(String security) {
+		this.security = security;
+	}
+
+	public void setBuyerBrokerId(String brokerId) {
+		this.buyerBrokerId = brokerId;
+	}
+	
+	public void setSellerBrokerId(String brokerId) {
+		this.sellerBrokerId = brokerId;
+	}
+
+	public void setDateTime(String date, String time) {
+		// Date and time format provided is 2013020500:00:00.000
+		SimpleDateFormat sicraOrderDateFormat = new SimpleDateFormat("yyyyMMddHH:mm:ss.SSS");
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The SimpleDateFormat object was created without setting the time zone. Make sure that you want to use the default time zone or use letter z, Z or X in the pattern. Otherwise, if you do not call setTimeZone() on the created SimpleDateFormat object, you might get an unexpected date and time when using the object.